jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -670,14 +667,33 @@ YUI.add('user-profile', function() {
     },
 
     /**
+      Handle the destroy model button being clicked.
+
+      @method _handleDestroyModel
+    */
+    _handleDestroyModel: function() {
+      var model = this.state.destroyModel;
+      var uuid = model.uuid;
+      // Hide the confirmation popup.
+      this._displayConfirmation(null);
+      // If the model to be distroyed is active then we can destroy it
+      // immediately.
+      if (uuid === this.props.currentModel) {
+        this._destroyModel(this.props.env);
+      } else {
+        // Switch to the selected model, then destroy it.
+        this.switchModel(uuid, model.name, this._destroyModel);
+      }
+    },
+
+    /**
       Destroy a model.
 
       @method _destroyModel
-      @return {Object} The confirmation component.
+      @param {Object} env The current env.
     */
-    _destroyModel: function() {
-      this._displayConfirmation(null);
-      this.props.destroyModel((error) => {
+    _destroyModel: function(env) {
+      env.destroyModel((error) => {
```

Comment:
If there is no error, should we notify that the model has been successfully destroyed?